packages/shared_preferences/shared_preferences_platform_interface/lib/method_channel_shared_preferences.dart

```diff
@@ -38,6 +38,20 @@ class MethodChannelSharedPreferencesStore
     return (await _kChannel.invokeMethod<bool>('clear'))!;
   }
 
+  @override
+  Future<Map<String, Object>> getAllWithPrefix(String prefix) async {
```

Comment:
FWIW, adding new functionality to default method channel implementations is optional; the idea is that they are to avoid breaking legacy 3P implementations that relied on them, and haven't updated to the new federated structure—and legacy implementations won't have new functionality.

It's fine to do it, but definitely not required.